agent/stats.go

```diff
@@ -112,6 +122,17 @@ func (s *statsReporter) reportLocked(
 	s.L.Unlock()
 	defer s.L.Lock()
 	stats := s.collector.Collect(ctx, networkStats)
+
+	// if the experiment is enabled we zero out certain session stats
+	// as we migrate to the client reporting these stats instead.
```

Comment:
Can this get implemented serverside instead? It seems like a lot of changes in the agent for something that could be zeroed out in the stats API on the server.

Reply:
Not really and here's why - If customers update coder but does not rebuild the workspace, we will have an old agent client and old cli version in the workspace. If the experiment is enabled and we zero out the value server side, we will lose data for that workspace because the old cli will not be reporting the new data. So the graphs will go to zero and slowly come back up over time as each workspace is restarted. 

So we need the endpoint to still save this data for old workspaces, but also allow new workspaces to report it in the new way. Or we accept downtime on the stats on upgrade until all workspaces are updated. Thoughts?

Reply:
> If the experiment is enabled and we zero out the value server side, we will lose data for that workspace because the old cli will not be reporting the new data via the usage endpoint.

Won't the same thing happen even if the feature was GA? Old agent/CLI versions need to be supported on new servers for multiple months.

This will also affect stats collection for older CLIs being used outside of workspaces, which arguably is where more people use the CLI anyways. I don't think anyone is using the CLI in a workspace to connect to a workspace.

> So we need the endpoint to still save this data for old workspaces, but also allow new workspaces to report it in the new way. Or we accept downtime on the stats on upgrade until all workspaces are updated.

I think either way there will be a "downtime" of stats with or without the agent changes when an older CLI is being used on the local machine.

I think this PR should change into just the CLI stats upload portion and have both sides report stats for now. Then in a few months you can remove stats reporting from the API by deprecating the field and ignoring it serverside.